### PR TITLE
Implements github create event and tag support

### PR DESCRIPTION
**Pull request by cbron**
Issue: https://github.com/rancher/rio/issues/542

### CODE REVIEW

File: pkg/provider/github/github.go [hunk 1]
```diff
@@ -238,12 +239,37 @@ func (w *GitHub) HandleHook(ctx context.Context, req *http.Request) (int, error)
...
+		if err != nil {
+			return http.StatusUnprocessableEntity, err
+		}
+		if parsed.Sender != nil {
+			execution.Spec.Author = safeString(parsed.Sender.Login)
+			execution.Spec.AuthorEmail = safeString(parsed.Sender.Email)
+			execution.Spec.AuthorAvatar = safeString(parsed.Sender.AvatarURL)
+		}
+
 	case *github.PushEvent:
```
Comment (StrongMonkey) [inline pkg/provider/github/github.go:264]: Does push event still receive tags? I thought we decided that push only watches commits while create watches tags.

Comment (cbron) [reply pkg/provider/github/github.go:264]: Push includes new tags yes, even via tag creation in UI. Create watches both tags and new branches so I have to exclude branches.

Comment (cbron) [reply pkg/provider/github/github.go:264]: fixed

File: pkg/provider/github/github.go [hunk 2]
```diff
@@ -238,13 +239,35 @@ func (w *GitHub) HandleHook(ctx context.Context, req *http.Request) (int, error)
 func (w *GitHub) handleEvent(ctx context.Context, client *github.Client, event interface{}, receiver *webhookv1.GitWatcher) (int, error) {
 	execution := initExecution(receiver)
 	switch event.(type) {
+	case *github.CreateEvent:
+		if receiver.Spec.Tag == false {
+			return http.StatusUnprocessableEntity, fmt.Errorf("create events only support tag")
```
Comment (StrongMonkey) [inline pkg/provider/github/github.go]: Not likely to happen. (if tag is false you won't get create event) Can we change WatchOnTag is not turned on?

File: pkg/provider/github/github.go [hunk 3]
```diff
@@ -238,13 +239,35 @@ func (w *GitHub) HandleHook(ctx context.Context, req *http.Request) (int, error)
...
+		}
+		parsed := event.(*github.CreateEvent)
+		if parsed.Ref == nil {
+			return http.StatusUnprocessableEntity, errors.New("create event has invalid tag ref")
```
Comment (StrongMonkey) [inline pkg/provider/github/github.go]: `create event has empty ref` would be more precise?

**Review by StrongMonkey — APPROVED** [review]
LGTM, minor nits.

File: pkg/provider/github/github.go (hunk 2 above)
Comment (cbron) [reply pkg/provider/github/github.go]: updated

File: pkg/provider/github/github.go (hunk 3 above)
Comment (cbron) [reply pkg/provider/github/github.go]: updated

File: pkg/apis/gitwatcher.cattle.io/v1/types.go [hunk 4]
```diff
@@ -31,8 +31,10 @@ type GitWatcherSpec struct {
 	Provider                       string            `json:"provider,omitempty"`
 	Push                           bool              `json:"push,omitempty"`
 	PR                             bool              `json:"pr,omitempty"`
-	Tag                            bool              `json:"tag,omitempty"`
 	Branch                         string            `json:"branch,omitempty"`
+	Tag                            bool              `json:"tag,omitempty"`
+	TagQuery                       string            `json:"tagQuery,omitempty"`
```
Comment (ibuildthecloud) [inline pkg/apis/gitwatcher.cattle.io/v1/types.go]: What about TagInclude or TagIncludeRegexp and TagExcludeRegexp.  Just seems like the two regexp's should be similar in name.

Comment (ibuildthecloud) [reply pkg/apis/gitwatcher.cattle.io/v1/types.go]: Or TagRegexp, TagExcludeRegexp.

Comment (cbron) [reply pkg/apis/gitwatcher.cattle.io/v1/types.go]: updated

**Pull request merged**